hathor/manager.py

```diff
@@ -419,27 +419,22 @@ def _initialize_components(self) -> None:
             try:
                 if self._full_verification:
                     # TODO: deal with invalid tx
-                    if self.tx_storage.is_tx_needed(tx.hash):
```

Comment:
Because this changes were made on top of improved-reward-lock + sync-v2 where `tx._height_cache` is not needed anymore. When I ported the changes back to the current `dev` branch, I also removed `tx._height_cache` references.